packages/@okta/vuepress-site/docs/reference/hook-best-practices/index.md

```diff
@@ -0,0 +1,67 @@
+
+  * Using [HTTP Basic Authentication](/books/api-security/authn/api-authentication-options/#http-basic-authentication) in the format `base64(user:password)`. See [Adding Basic Authorization and Body Parsing](/docs/guides/common-hook-set-up-steps/) for an example implementation of Basic Authentication.
+  * Using a separate Secret Key, which can be configured on the Hook as a custom header field.
+
+* Create an allowlist of IP addresses to check on incoming Okta calls. See the following Okta Knowledge Base article for a listing of Okta IP addresses: [What IP addresses/ranges should we allow for inbound traffic?](https://support.okta.com/help/s/article/What-IP-addresses-ranges-should-we-whitelist-for-inbound-traffic-i-e-REST-API-calls-from-Okta-to-on-prem-JIRA-server?language=en_US).
+
+>**Note:** This is a large list of IP Addresses and the list is subject to change. Unless required by your organization, securing your Hook by Authentication header is recommended.
+
+## Protecting your Hook content from external viewers
+
+To prevent unauthorized parties from reading the contents of an Okta Hook, we recommend the following:
+
+* Okta uses HTTPS to encrypt communications to your Hook endpoint. When using HTTPS, ensure you keep your SSL certificate updated and the Domain Name System (DNS) secured, so that someone can’t re-point your endpoint to another location.
+
+## Avoiding delays in Inline Hook responses
+
+When Okta uses an Inline Hook to communicate with your endpoint, the user experience is paused until your code responds. To prevent unnecessary delays or timeouts, we recommend the following:
+
+* Treat Okta Hooks as you would any other HTTP request, namely: Ensure you respond to the HTTP request in under 400 milliseconds with a 200 (Success) or 204 (Success no content) code.
+
+>**Note:** The Okta request times out after 3 seconds with no response for Inline Hooks and Event Hooks.
+
+See Inline Hook [Timeout and Retry](/docs/concepts/inline-hooks/#timeout-and-retry) for further information on Inline Hook timeout and retry behavior; see Event Hook [Timeout and Retry](https://developer.okta.com/docs/concepts/event-hooks/) for the same.
+
+## Working with Event Types (?)
+
+* Do we have any guidelines on the number of Event Types we can add to a hook? Our current documentation suggests we can configure "multiple event types". Would it be better to have multiple endpoints for multiple event types, from a load perspective? Or does it matter?
+
+## Managing scale of Okta Hook calls (?)
+
+* Any advice necessary on this subject, which I've seen on other webhhook pages. E.g. "If you’re on the other side running a system that is consuming webhooks, you can scale your webhook ingestion the same way you’d horizontally scale for regular web traffic — by using a load balancer or reverse-proxy in front of your web servers. "
+
+## Ordering and Duplicates of Hooks (?)
```

Comment:
We don't guarantee ordering of hooks but if they want to avoid duplicate processing of Event Hooks, they can store  the "eventId" field of an event hook and only process event hooks they haven't received yet based on the eventId